### ISSUE TITLE
Investigate strategies to fully implement the POSIX [=equivalence-class=] syntax.

### ISSUE THREAD

**Issue by mikehaertel** [mikehaertel/minrx#15]

**Comment by bhaible** [comment]
Gnulib has the same problem; see the code starting at gnulib/lib/regexec.c line 3774.
IMO we should ask the glibc people for a libc API for this.

**Comment by arnoldrobbins** [comment]
@bhaible A GLIBC API is necessary but not sufficient (as my high school math teacher used to say). We need a POSIX API in order to be able to work portably on macOS and legacy Unixes (and maybe even Windows).

**Comment by bhaible** [comment]
> We need a POSIX API

Right. But the Austin Group rarely standardizes an API that no libc has implemented so far. Therefore the most promising approach for getting a POSIX API is to get a glibc API first, and then (with the help of Eric Blake) push that onto the Austin Group's table.

**Comment by arnoldrobbins** [comment]
See #30 for a suggested implementation that works and is good enough for now.

**Comment by arnoldrobbins** [comment]
Can this be closed now that #30 is merged? Thanks.

**Comment by mikehaertel** [comment]
I am leaving this open to remind us that we need a better solution than searching 1.1M+ codepoints every time an equivalence class is used...

**Comment by arnoldrobbins** [comment]
Looking at GNULIB regcomp.c, the support for equivalence classes in GLIBC is only good for single byte characters. So I wonder if we can even get the info we need out of GLIBC for multiybte characters?  I'm starting to think we might need to roll our own somehow.

Also, as I've noted before, in practice I don't think anyone uses equivalence classes.  I've *never* had a complaint about the lack of support for them in gawk.  So maybe we're overthinking this?

**Comment by arnoldrobbins** [comment]
@mikehaertel I think this is solved in the current main-c branch. At least for Unicode locales.
